### ISSUE TITLE
10.0.0 introduces [ENOTSUP] NO_ABORT_CONTROLLER on Node 14 (latest)?

### ISSUE THREAD

**Issue by DavidAnson** [isaacs/node-glob#522]
I lack the means to investigate at the moment (sorry), but the Dependabot PR for glob 9.3.4 to 10.0.0 fails CI on Node 14 (only) for the markdownlint-cli project:

https://github.com/igorshubovych/markdownlint-cli/pull/384

Every test fails with the following output:

```
  - `(node:1802) [ENOTSUP] NO_ABORT_CONTROLLER: AbortController is not defined. If using lru-cache in node 14, load an AbortController polyfill from the \`node-abort-controller\` package. A minimal polyfill is provided for LRUCache.fetch(), but this may cause undesirable behavior in other APIs that use AbortController.
```

glob's package.json still declares support for latest Node 14, so this seems unexpected?

PS: Parallel CI run to demonstrate the base branch of the PR passes on all platforms: https://github.com/igorshubovych/markdownlint-cli/actions/runs/4670376350

**Comment by DavidAnson** [comment]
I should note that while this IS a semver-breaking change for glob, the change log only calls out "No default exports, only named exports".

**Comment by isaacs** [comment]
It's only "breaking" if you have Node configured to throw on warnings.

I can make the lru-cache warning even more buried, so it only prints if you actually _use_ cache.fetch, I suppose?

**Comment by isaacs** [comment]
But in any event, it's (a) a completely valid warning, you are using code that might behave oddly if not provided with the appropriate polyfill (and it is hard to argue that warning is worse than _not_ warning, for an issue that was already present), and (b) the remediation is fairly straightforward.

So, no, this is not a breaking change, it is a bugfix.

**Comment by DavidAnson** [comment]
Sorry, I was not clear. I was saying that the version change from 9 to 10 allowed for breaking changes in glob, so it was not wrong for this change to be present. However, because it introduces failures to passing CI runs in the default GitHub configuration, I suggest the impact of this change is breaking-like. :) I am not arguing with your assessment, but I suggest that including this in the change log would be helpful for others.

**Comment by isaacs** [comment]
Ah, ok.  Sorry if my pushback was a little too forceful then, I misinterpreted 😅

> I suggest that including this in the change log would be helpful for others.

Not a terrible suggestion, but I think this is even better: I just updated lru-cache so that it'll only print the warning if `cache.fetch()` is actually used.  That's way less disruptive anyway.

`npm update lru-cache` and verify that `npm ls lru-cache` shows v 9.0.3, and I think this ends up being resolved.
